### DISCUSSION TITLE
loading data in solid stream example

### DISCUSSION THREAD

**Discussion by MadadiSaeed** [solidjs/solid#821] [Q&A] [answered]
- In the solid stream example how to connect actual data from server/database to the app when **rendering** (calling **reanderToStream**)? The solution that come to my mind is to pass app data as root component's (**App** component) props and use it as initial value of **App** context or any context provider. Is there any other solution?  [https://github.com/solidjs/solid/tree/main/packages/solid-ssr/examples/stream](https://github.com/solidjs/solid/tree/main/packages/solid-ssr/examples/stream)
- How to load data from server inside **createResource** function? [https://github.com/solidjs/solid/blob/main/packages/solid-ssr/examples/shared/src/components/Profile/index.js](https://github.com/solidjs/solid/blob/main/packages/solid-ssr/examples/shared/src/components/Profile/index.js)

**Comment by ryansolid** [comment, marked as answer]
Yeah this example uses isomorphic data fetching, in that it mocks an API that would be called from the client and server. To have this work straight from a database takes some more work since the browser can't speak to it directly to the database so you need an API. In an ideal world you'd do the database call directly on the server and through the API in the browser. This takes some special wiring with `createResource` and use of `isServer` from `solid-js/web` to decide which to execute in browser/server.

`createResource` takes an async function so it really can be anything that you fetch. It can be an API db or whatnot. Just have it return the promise with the data and Solid with conjunction with Suspense. But the code runs on both browser and server so it isn't just as simple as calling the db query in there.

We are working on a metaframework (solid-start) to make these sort of things easier. But it is still under construction.
